packages/core/src/html/ComponentProcessor.js

```diff
@@ -0,0 +1,162 @@
+const cheerio = require('cheerio');
+
+const { findHeaderElement } = require('./headerProcessor');
+const { getVslotShorthandName } = require('./vueSlotSyntaxProcessor');
+const { warnConflictingAttributes, warnDeprecatedAttributes } = require('./warnings');
+
+const _ = {};
+_.has = require('lodash/has');
+
+const logger = require('../utils/logger');
+
+class ComponentProcessor {
```

Comment:
Hmm should we split this up further? Namely
- conflicting attribute logging
- deprecated attribute logging (can be fully removed in a prior pr also though, if easier https://github.com/MarkBind/markbind/issues/1504)
- the katex one can be further split with e.g. the one for `<pre>` (and any others) into something like `vPreAdder.js` (or just in the switch-case since its pretty short)

So this file / class would tackle just rendering markdown-in-attributes

Reply:
to further add, my pov is that `NodeProcessor` is still and should be the one doing the 'organizing by elements', so it may make more sense to tie up / order various functionalities required by those elements there anyway. (instead of pushing it one level deeper down to somewhere like `BootstrapVueProcessor`)

Reply:
> conflicting attribute logging
deprecated attribute logging (can be fully removed in a prior pr also though, if easier #1504)

I have created a mapping of `node.name` to functions that does the warnings for conflicting attributes and deprecated attributes in `warnConflictingAtributesMap` and `warnDeprecatedAtributesMap` respectively in `warning.js`. And the various warnings are called before the switch-case statements of `NodeProcessor#processNode`. I thought that this is quite simple and easy for someone to easily add / remove warnings for various components via the mapping. What are your thoughts on this?

> the katex one can be further split with e.g. the one for `<pre>` (and any others) into something like vPreAdder.js (or just in the switch-case since its pretty short)

Since the logic is pretty short, I have added it in the switch-case.

> to further add, my pov is that NodeProcessor is still and should be the one doing the 'organizing by elements', so it may make more sense to tie up / order various functionalities required by those elements there anyway. (instead of pushing it one level deeper down to somewhere like BootstrapVueProcessor)

I may need some clarifications on this. Based on what I understand, you are saying `NodeProcessor` should have some utility functions which are used by all elements? 

From `BootstrapVueProcessor`, I have moved `_renameSlot` and `_renameAttributes` to `NodeProcessor` since it is quite general and not really specific to the functionality of `BootstrapVueProcessor`. However, I feel that it be may better for certain functions such as `_addTriggerClass` and `_transformSlottedComponents` to be in `BootstrapVueProcessor` as they are helper functions for performing the bootstrap-vue transformations.  Do let me know if I am misunderstanding this.

Reply:
> What are your thoughts on this?

Looks good, even cleaner now. 👍 

> I may need some clarifications on this. Based on what I understand, you are saying NodeProcessor should have some utility functions which are used by all elements?

Sorry, no action needed here. Just a remark on what I think `NodeProcessor` should be handling (consolidating and ordering functionalities from the separate files for each element)

> `ComponentProcessor`

Should we rename the class as such? Maybe `MdAttributeRenderer`? 😮

Reply:
> Should we rename the class as such? Maybe `MdAttributeRenderer`? 😮

Yes I have no issues with this name ^_^